added_values.go

```diff
@@ -0,0 +1,608 @@
+}
+
+func (i valueUInt32) Equals(other Value) bool {
+	if o, ok := other.assertInt32(); ok {
+		return int32(i) == o
+	}
+	if o, ok := other.assertFloat(); ok {
+		return float64(i) == o
+	}
+	if o, ok := other.assertString(); ok {
+		return o.ToNumber().Equals(i)
+	}
+	if o, ok := other.(valueBool); ok {
+		return int(i) == o.ToInt()
+	}
+	if o, ok := other.(*Object); ok {
+		return i.Equals(o.self.toPrimitiveNumber())
+	}
+	return false
+}
+
+func (i valueUInt32) StrictEquals(other Value) bool {
+	if otherInt, ok := other.assertInt32(); ok {
+		return int32(i) == otherInt
+	} else if otherFloat, ok := other.assertFloat(); ok {
+		return float64(i) == otherFloat
+	}
+	return false
+}
+
+func (i valueUInt32) hash(*maphash.Hash) uint64 {
+	return uint64(i)
+}
+
+func (i valueUInt32) assertInt() (int, bool) {
+	return 0, false
+}
+func (i valueUInt32) assertUInt32() (uint32, bool) {
+	return uint32(i), true
+}
+func (i valueUInt32) assertInt32() (int32, bool) {
+	return 0, false
+}
+func (i valueUInt32) assertInt64() (int64, bool) {
+	return 0, false
+}
+
+func (i valueUInt32) assertFloat() (float64, bool) {
+	return 0, false
+}
+
+func (i valueUInt32) assertString() (valueString, bool) {
+	return nil, false
+}
+
+func (i valueUInt32) baseObject(r *Runtime) *Object {
+	return r.global.NumberPrototype
+}
+
+func (i valueUInt32) Export() interface{} {
+	return uint32(i)
+}
+
+func (i valueUInt32) ExportType() reflect.Type {
+	return reflectTypeUInt32
+}
+
+func (i valueInt32) MemUsage(ctx *MemUsageContext) (uint64, error) {
+	return NumberSize, nil
+}
+
+func (i valueInt32) hash(*maphash.Hash) uint64 {
+	return uint64(i)
+}
+
+func (i valueInt32) ToInt() int {
+	return int(i)
+}
+func (i valueInt32) ToInteger() int64 {
+	return int64(i)
+}
+
+func (i valueInt32) ToInt32() int32 {
+	return int32(i)
+}
+func (i valueInt32) ToUInt32() uint32 {
+	return uint32(i)
+}
+
+func (i valueInt32) ToInt64() int64 {
+	return int64(i)
+}
+
+func (i valueInt32) ToString() Value {
+	return asciiString(i.String())
+}
+func (i valueInt32) toString() valueString {
+	return asciiString(i.String())
+}
+func (i valueInt32) String() string {
+	return strconv.FormatInt(int64(i.ToInt()), 10)
+}
+
+func (i valueInt32) string() unistring.String {
+	return unistring.String(strconv.FormatInt(int64(i.ToInt()), 10))
+}
+func (i valueInt32) IsObject() bool {
+	return false
+}
+func (i valueInt32) IsNumber() bool {
+	return true
+}
+func (i valueInt32) ToFloat() float64 {
+	return float64(int64(i))
+}
+
+func (i valueInt32) ToBoolean() bool {
+	return i != 0
+}
+
+func (i valueInt32) ToObject(r *Runtime) *Object {
+	return r.newPrimitiveObject(i, r.global.NumberPrototype, classNumber)
+}
+
+func (i valueInt32) ToNumber() Value {
+	return i
+}
+
+func (i valueInt32) SameAs(other Value) bool {
+	if otherInt, ok := other.assertInt32(); ok {
+		return int32(i) == otherInt
+	}
+	return false
+}
+
+func (i valueInt32) Equals(other Value) bool {
+	if o, ok := other.assertInt32(); ok {
```

Comment:
This function looks identical to the `(i valueUInt32) Equals` function. Can me refactor this into a function called by both?